overlord/snapshotstate/snapshotstate.go

```diff
@@ -541,9 +565,42 @@ func Forget(st *state.State, setID uint64, snapNames []string) (snapsFound []str
 	return summaries.snapNames(), ts, nil
 }
 
+// setSnapshotOpInProgress marks the given set ID as being a subject of
+// snapshot op inside state cache. State must be locked by the caller.
+func setSnapshotOpInProgress(st *state.State, setID uint64, op string) {
+	var snapshotOps map[uint64]string
+	if val := st.Cached("snapshot-ops"); val != nil {
+		snapshotOps, _ = val.(map[uint64]string)
+	} else {
+		snapshotOps = make(map[uint64]string)
+	}
+	snapshotOps[setID] = op
+	st.Cache("snapshot-ops", snapshotOps)
+}
+
+func UnsetSnapshotOpInProgress(st *state.State, setID uint64) {
```

Comment:
this maybe should return a bool or the op to indicate if there was something to remove to help with tests ?